src/block/Block.php

```diff
@@ -634,4 +634,19 @@ public function calculateIntercept(Vector3 $pos1, Vector3 $pos2) : ?RayTraceResu
 
 		return $currentHit;
 	}
+
+	/**
+	 * Return items dropped by the block when it is broken by a player in gamemode creative
+	 * @return Item[]
+	 */
+	public function getCreativeDrops(): array{
+		return [];
+	}
+
+	/**
+	 * Return xp dropped by the block when it is broken by a player in gamemode creative
+	 */
+	public function getCreativeXpDrop(): int{
```

Comment:
It is true that the name is not very appropriate, however it is more related to the furnace when you break them, these drop the xp that is stored there.
Certainly this functionality is not yet implemented, but it is good to foresee it?
and above all, it just reflects the current vanilla functionality.

Reply:
This can already be implemented a different way anyway (see getXpDropAmount(), used by ores).

XP in creative is pointless, so I don't see any reason to add this.

Reply:
I think it is still necessary to implement the xp drop even if you are in creative. Why? I would already say that it allows as I said, the furnace to be able to drop the xp that is stored in it when the feature will be impelmenter, and then also because minecraft vanilla does it.
Maybe implementing a new feature is not necessary but integrating the feature is.

Reply:
I don't really care if vanilla does it, because vanilla does a bunch of stupid things that don't make sense.

It _does_ matter that it should be dropped in survival (where it actually matters) but there are already systems for doing that anyway.

Reply:
![image](https://user-images.githubusercontent.com/54852588/146226849-b321bc62-15d4-4d4e-bcaa-5ee6d0c9720c.png)
I agree, although furnaces currently don't produce xp during smelting, I think that this method would be useful so that we later can differentiate between breaking an ore and a furnace in creative

Reply:
I don't see why it makes any difference.

No one cares about XP in creative mode, and we already have `getXpDropAmount()`. There's no sense in adding an extra API method for this useless "feature" (which is most likely a bug anyway).

Reply:
> I don't really care if vanilla does it, because vanilla does a bunch of stupid things that don't make sense.
> 
> It _does_ matter that it should be dropped in survival (where it actually matters) but there are already systems for doing that anyway.

I'm beginning to agree indeed. Looking at the already existing api, it is true that we can already do this

Reply:
It's up to debate whether that is a bug or an intended feature and it seems that it worked for quite a while like that.
If dropping experience in creative mode would be added, we would need a creative mode method for `getXpDropAmount()`, since we can't use the normal one and just call it always, not just when the player is in survival mode, at that would result in ores and other blocks dropping xp when broken in creative.

> No one cares about XP in creative mode

That is true, but then, why should anyone care that the items of containers drop when in creative?

Reply:
For the items in the containers, this is really useful and makes sense

Reply:
> That is true, but then, why should anyone care that the items of containers drop when in creative?

Because items are user-created data, e.g. you might write a book and keep it in a chest.